watertap/examples/flowsheets/MD/tests/test_MD.py

```diff
@@ -0,0 +1,136 @@
+        assert isinstance(m.fs.mixer, Mixer)
+        assert isinstance(m.fs.heater, Heater)
+        assert isinstance(m.fs.chiller, Heater)
+        assert isinstance(m.fs.hx, HeatExchanger)
+        assert isinstance(m.fs.MD, MembraneDistillation0D)
+        assert isinstance(m.fs.separator_concentrate, Separator)
+        assert isinstance(m.fs.separator_permeate, Separator)
+        assert isinstance(m.fs.permeate, Product)
+
+        # additional variables
+        assert isinstance(m.fs.overall_recovery, Var)
+        assert isinstance(m.fs.recycle_ratio, Var)
+
+        # arcs
+        arc_dict = {
+            m.fs.s01: (m.fs.feed.outlet, m.fs.pump_feed.inlet),
+            m.fs.s02: (m.fs.pump_feed.outlet, m.fs.mixer.feed),
+            m.fs.s03: (m.fs.mixer.outlet, m.fs.hx.cold_inlet),
+            m.fs.s04: (m.fs.hx.cold_outlet, m.fs.pump_brine.inlet),
+            m.fs.s05: (m.fs.pump_brine.outlet, m.fs.heater.inlet),
+            m.fs.s06: (m.fs.heater.outlet, m.fs.MD.hot_ch_inlet),
+            m.fs.s07: (m.fs.MD.hot_ch_outlet, m.fs.separator_concentrate.inlet),
+            m.fs.s08: (m.fs.separator_concentrate.reject, m.fs.reject.inlet),
+            m.fs.s09: (m.fs.separator_concentrate.recycle, m.fs.mixer.recycle),
+            # pemeate (MD cold side) loop
+            m.fs.s10: (m.fs.chiller.outlet, m.fs.MD.cold_ch_inlet),
+            m.fs.s11: (m.fs.MD.cold_ch_outlet, m.fs.hx.hot_inlet),
+            m.fs.s12: (m.fs.hx.hot_outlet, m.fs.separator_permeate.inlet),
+            m.fs.s13: (m.fs.separator_permeate.permeate, m.fs.permeate.inlet),
+            m.fs.s14: (
+                m.fs.separator_permeate.cold_loop_stream,
+                m.fs.pump_permeate.inlet,
+            ),
+            m.fs.s15: (m.fs.pump_permeate.outlet, m.fs.chiller.inlet),
+        }
+        for arc, port_tpl in arc_dict.items():
+            assert arc.source is port_tpl[0]
+            assert arc.destination is port_tpl[1]
+
+    @pytest.mark.component
+    def test_units(self, system_frame):
+        assert_units_consistent(system_frame)
+
+    @pytest.mark.component
+    def test_set_operating_conditions(self, system_frame):
+        m = system_frame
+        set_operating_conditions(m)
+        assert degrees_of_freedom(m) == 0
+
+    @pytest.mark.component
+    @pytest.mark.requires_idaes_solver
+    def test_initialize_system(self, system_frame):
+        m = system_frame
+        initialize_system(m, solver=solver)
+        assert degrees_of_freedom(m) == 0
+
+    @pytest.mark.requires_idaes_solver
+    @pytest.mark.component
+    def test_solution(self, system_frame):
+        m = system_frame
+        optimize_set_up(m)
+        solve(m, solver=solver)
+        assert pytest.approx(0, rel=1e-5) == value(
+            m.fs.permeate.flow_mass_phase_comp[0, "Liq", "TDS"]
+        )
+        assert pytest.approx(0.5, rel=1e-5) == value(m.fs.overall_recovery)
+        assert pytest.approx(6.169, rel=1e-3) == value(m.fs.recycle_ratio[0])
```

Comment:
In this test, it would be good to check more of the key results that are calculated. For example, you can check `LCOW`, `specific_energy_consumption`, etc.)

Reply:
done